test/cql-pytest/random_util.py

```diff
@@ -0,0 +1,178 @@
+# randomized elements allow to have great coverage with time.
+#
+# Use fixture `randomizer_seed` present in conftest.py, to print
+# seed used for generation. If a test fails, this seed can be used to
+# reproduce the test.
+
+import random
+from contextlib import contextmanager
+from util import new_type, unique_name, new_test_table, new_test_keyspace
+from pytest import fixture
+
+# When generating random type if depth > max_depth
+# then force native_type
+max_depth = 2
+# How many types can be in collection/tuple/UDT
+max_types = 5
+# How many rows can be in partition key
+max_pk = 3
+# How many rows can be in clustering key
+max_ck = 5
+# How many regular rows cam be in table
+max_regular = 4
+# How many static rows cam be in table
+max_static = 3
+# Chance of counter table instead of normal one
+counter_table_chance = 0.1
+
+# A utility function for obtaining random native type
+def get_random_native_type():
+    types = ["text", "bigint", "boolean", "decimal", "double", "float", "int"]
+    return random.choice(types)
+
+# A utility function for obtaining random collection type.
+# `udts` argument represents user-defined types that can be used for
+# creating the collection type. If `frozen` is true, then the function will
+# return `frozen<collection_type>`
+def get_random_collection_type(cql, keyspace, udts=[], frozen=False, depth=0):
+    collection_type = random.randrange(3)
+
+    return_type = ""
+    if collection_type == 0:
+        return_type = f"set<{get_random_type(cql, keyspace, udts, True, depth+1)}>"
+    elif collection_type == 1:
+        return_type = f"list<{get_random_type(cql, keyspace, udts, True, depth+1)}>"
+    elif collection_type == 2:
+        return_type = f"map<{get_random_type(cql, keyspace, udts, True, depth+1)}, {get_random_type(cql, keyspace, udts, True, depth+1)}>"
+    return f"frozen<{return_type}>" if frozen else return_type
+
+# A utility function for obtaining random tuple type.
+# `udts` argument represents user-defined types that can be used for
+# creating the tuple type. If `frozen` is true, then the function will
+# return `frozen<tuple<some_fields>>`
+def get_random_tuple_type(cql, keyspace, udts=[], frozen=False, depth=0):
+    n = random.randrange(1, max_types+1)
+    types = [get_random_type(cql, keyspace, udts, False, depth+1) for _ in range(n)]
+    fields = ", ".join(types)
+    return_type = f"tuple<{fields}>"
+
+    return f"frozen<{return_type}>" if frozen else return_type
+
+# A utility function for obtaining random type. The function can return
+# native types, collections, tuples and UDTs. `udts` argument represents
+# user-defined types that can be returned or used for creation of other types
+# If `frozen` is true, then the function will return `frozen<some_type>`
+def get_random_type(cql, keyspace, udts=[], frozen=False, depth=0):
+    type_type = random.randrange(4 if len(udts) != 0 else 3)
+
+    if type_type == 0 or depth > max_depth: # native type
```

Comment:
So that "max_depth" doesn't need to be a global variable, you can change all these functions to **start** with max_depth (a parameter), and decrement it until it reaches zero - instead of incrementing as you do here and relying on a global "max_depth" for stopping.